### PR TITLE
Call daemon-reload after we touched systemd unit files (LP: #1874494)

### PR DESCRIPTION
**Pull request by slyon**
##  Description
When running `netplan apply` systemd unit files might be created/deleted/changed on disk. We need to run `systemctl daemon-reload`, to make systemd aware of those new units and calculate the new unit/service dependencies. Otherwise it will throw warnings at us:
```
ubuntu@ubuntu:~$ sudo netplan apply
Warning: The unit file, source configuration file or drop-ins of netplan-wpa-wlan0.service changed on disk. Run 'systemctl daemon-reload' to reload units.
Warning: The unit file, source configuration file or drop-ins of netplan-wpa-wlan0.service changed on disk. Run 'systemctl daemon-reload' to reload units.
```

Fixes: https://bugs.launchpad.net/ubuntu/+source/netplan.io/+bug/1874494

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains 100% code coverage (`make check-coverage`).
- [ ] New/changed keys in YAML format are documented.
- [x] \(Optional\) Closes an open bug in Launchpad.

### CODE REVIEW

**Comment by slyon** [comment]
Integration tests looking good! Also, I verified it manually on RaspberryPi, all OK.

File: netplan/cli/commands/apply.py [hunk 1]
```diff
@@ -108,6 +108,10 @@ def command_apply(run_generate=True, sync=False, exit_on_error=True):  # pragma:
         # stop backends
         if restart_networkd:
             logging.debug('netplan generated networkd configuration changed, restarting networkd')
+            # Running 'systemctl daemon-reload' will re-run the netplan systemd generator,
+            # so let's make sure we only run it iff we're willing to run 'netplan generate'
```
Comment (sil2100) [inline netplan/cli/commands/apply.py:112]: Typo, 'iff'.

Comment (slyon) [reply netplan/cli/commands/apply.py:112]: 'iff' was used on purpose, with the meaning of 'if and only if'.

**Comment by xnox** [comment]
When wifi was a template @.service unit, we didn't change any units.

**Pull request merged**